lldb/source/Core/Progress.cpp

```diff
@@ -30,18 +34,19 @@ Progress::~Progress() {
   // Make sure to always report progress completed when this object is
   // destructed so it indicates the progress dialog/activity should go away.
   std::lock_guard<std::mutex> guard(m_mutex);
-  if (!m_completed)
-    m_completed = m_total;
+  if (m_total.has_value() && !m_completed)
+    m_completed = m_total.value();
```

Comment:
revert (see comments above and down in `Progress::ReportProgress(...)` for details

Reply:
Seconded from your comments on `ReportProgress` above.